format/FlightSql.proto

```diff
@@ -1450,8 +1512,72 @@ message ActionClosePreparedStatementRequest {
   bytes prepared_statement_handle = 1;
 }
 
+/*
+ * Request message for the "BeginTransaction" action.
+ * Begins a transaction.
+ */
+message ActionBeginTransactionRequest {
+  option (experimental) = true;
+}
+
+/*
+ * Request message for the "BeginSavepoint" action.
+ * Creates a savepoint within a transaction.
+ *
+ * Only supported if FLIGHT_SQL_TRANSACTION is
+ * FLIGHT_SQL_TRANSACTION_SUPPORT_SAVEPOINT.
+ */
+message ActionBeginSavepointRequest {
+  option (experimental) = true;
+
+  // The transaction to which a savepoint belongs.
+  bytes transaction_id = 1;
+  // Name for the savepoint.
+  string name = 2;
+}
+
+/*
+ * The result of a "BeginTransaction" or "BeginSavepoint" action.
+ *
+ * The transaction can be manipulated with the "EndTransaction" action, or
+ * automatically via server timeout. If the transaction times out, then it is
+ * automatically rolled back.
+ *
+ * The result should be wrapped in a google.protobuf.Any message.
+ */
+message ActionBeginTransactionResult {
+  option (experimental) = true;
+
+  // Opaque handle for the transaction or savepoint on the server.
+  bytes transaction_id = 1;
+}
+
+/*
+ * Request message for the "EndTransaction" action.
+ *
+ * With a transaction: commit (COMMIT) or rollback (ROLLBACK) the transaction.
+ * With a savepoint: remove (COMMIT) or rollback to (ROLLBACK) the savepoint.
+ *
+ * If the action completes successfully, the transaction/savepoint handle is
+ * invalidated.
+ */
+message ActionEndTransactionRequest {
+  option (experimental) = true;
+
+  enum EndTransaction {
+    END_TRANSACTION_UNSPECIFIED = 0;
+    // Commit the transaction or release the savepoint.
+    END_TRANSACTION_COMMIT = 1;
+    // Roll back the transaction or to a savepoint.
+    END_TRANSACTION_ROLLBACK = 2;
```

Comment:
It's weird that a savepoint rollback is in the end transaction request. It isn't ending the transaction, it's only rolling back intermediary progress, right?

Reply:
I rolled them together since I felt they were "effectively" the same, but I can break them into separate calls.

Reply:
No strong opinion. Just seemed weird with fresh eyes.

Reply:
I split them entirely just to avoid confusion (and so transaction/savepoint handles no longer have to share a namespace)